sherpa/astro/ui/utils.py

```diff
@@ -2417,8 +2455,9 @@ def set_filter(self, id, val=None, bkg_id=None, ignore=False):
         if val is None:
             val, id = id, val
 
-        d = self.get_data(id)
-        if bkg_id is not None:
+        if bkg_id is None:
+            d = self.get_data(id)
+        else:
             d = self.get_bkg(id, bkg_id)
```

Comment:
Since this 4-line pattern is to common, does it make sense to have something equivalent to `_get_pha_data(idval, bkg_id)` like the following?
```Python
def _get_data_or_bkg(idval, bkg_id):
    if bkg_id is None:
        d = self.get_data(id)
    else:
        d = self.get_bkg(id, bkg_id)
```

Reply:
I saw that comment, but I found it easier to say it where it's happening.